.github/workflows/checkstyle.yaml

```diff
@@ -13,32 +13,44 @@ jobs:
         ref: ${{ github.event.pull_request.head.sha }}
     - name: Install dependencies
       run: |
+        echo 'force-unsafe-io' | sudo tee /etc/dpkg/dpkg.cfg.d/force_unsafe_io
         sudo apt-get update
-        sudo apt-get install --yes -qq build-essential autoconf libtool gawk alien fakeroot linux-headers-$(uname -r)
-        sudo apt-get install --yes -qq zlib1g-dev uuid-dev libattr1-dev libblkid-dev libselinux-dev libudev-dev libssl-dev python3 python3-dev python3-setuptools python3-cffi
-        # packages for tests
-        sudo apt-get install --yes -qq parted lsscsi ksh attr acl nfs-kernel-server fio
-        sudo apt-get install --yes -qq mandoc cppcheck pax-utils devscripts
-        sudo -E pip --quiet install flake8
+        xargs --arg-file=./.github/workflows/build-dependencies.txt sudo apt-get install -qq
+        sudo apt-get install -qq mandoc cppcheck pax-utils devscripts
+        sudo python3 -m pip install --quiet flake8~=4.0.1
+
+        # abigail>=2.0 is available in `jammy` and later releases
+        echo 'deb http://azure.archive.ubuntu.com/ubuntu/ jammy main universe' | sudo tee -a /etc/apt/sources.list
+        sudo apt-get update
+        sudo apt-get install -t jammy -qq abigail-tools
+        sudo apt-get clean
```

Comment:
Given how often we've seen incompatible format changes made to libabigail, I think it would be reasonable to stick with our current container.  At least for the time being.

Reply:
+1, docker image is used not only here, let's leave it as is.

Reply:
I understand the concern, the main Makefile

```
checklibabiversion:
  libabiversion=`abidw -v | $(SED) 's/[^0-9]//g'`; \
  if test $$libabiversion -lt "200"; then \
        /bin/echo -e "\n" \
        "*** Please use libabigail 2.0.0 version or newer;\n" \                                                                                                                               
        "*** otherwise results are not consistent!\n" \
        "(or see https://github.com/openzfs/libabigail-docker )\n"; \
        exit 1; \
    fi;
```

The docker image is based on Ubuntu jammy:
```
$ podman run --rm -it ghcr.io/openzfs/libabigail cat /etc/os-release
PRETTY_NAME="Ubuntu Jammy Jellyfish (development branch)"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04 (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=impish
```

Jammy release is coming soon (~ 3 months) and it seems unlikely to me that there will be a major `abigail-tools` upgrade in there.

Reply:
@szubersk unfortunately, libabigail changes it's output regularly. That's why we have a fixed docker image to use, we generate new ABI files only via this image and non-Linux contributors use same image for this check too. So image usage in CI is mandatory for us not only because of Ubuntu minimal version now.

Reply:
Ok, understood.